custom_components/tuya_local/devices/kogan_lx10_vacuum.yaml

```diff
@@ -0,0 +1,259 @@
+name: Kogan LX10 Robot Vacuum
+products:
+  - id: ldb9ygikft5awhlu
+    name: Kogan LX10 Robot Vacuum
+primary_entity:
+  entity: vacuum
+  dps:
+    - id: 101
+      name: clean_switch
+      type: boolean
+    - id: 102
+      name: pause_switch
+      type: boolean
+    - id: 103
+      name: charge_switch
+      type: boolean
+    - id: 104
+      name: clean_mode
+      type: string
+      mapping:
+        - dps_val: smart
+          value: smart
+        - dps_val: pose
+          value: pose
+        - dps_val: zone
+          value: zone
+        - dps_val: backcharge
+          value: backcharge
+    - id: 105
+      name: robot_state
+      type: string
+      mapping:
+        - dps_val: idle
+          value: idle
+        - dps_val: pointing\areaing
+          value: pointing\areaing
+        - dps_val: totaling
+          value: totaling
+        - dps_val: sweep
+          value: sweep
+        - dps_val: mop
+          value: mop
+        - dps_val: fault
+          value: fault
+        - dps_val: pause
+          value: pause
+        - dps_val: chargring
+          value: chargring
+        - dps_val: tocharge
+          value: tocharge
+        - dps_val: fullcharge
+          value: fullcharge
+        - dps_val: remotectl
+          value: remotectl
+    - id: 106
+      name: battery
+      type: integer
+    - id: 107
+      name: cur_clean_time
+      type: integer
+    - id: 108
+      name: cur_clean_area
+      type: integer
+    - id: 109
+      name: fan_mode
+      type: string
+      mapping:
+        - dps_val: quiet
+          value: quiet
+        - dps_val: strong
+          value: strong
+        - dps_val: auto
+          value: auto
+    - id: 110
+      name: water_mode
+      type: string
+      mapping:
+        - dps_val: 'off'
+          value: 'off'
+        - dps_val: quiet
+          value: quiet
+        - dps_val: auto
+          value: auto
+        - dps_val: strong
+          value: strong
+    - id: 111
+      name: remote_ctrl
+      optional: true
+      type: string
+      mapping:
+        - dps_val: forward
+          value: forward
+        - dps_val: backward
+          value: backward
+        - dps_val: left
+          value: left
+        - dps_val: right
+          value: right
+        - dps_val: stop
+          value: stop
+    - id: 112
+      name: seek_robot
+      optional: true
+      type: boolean
+    - id: 113
+      name: disturb_switch_1
+      type: boolean
+    - id: 114
+      name: volume
```

Comment:
Split out into separate number or sensor entity under secondary entities, depending on whether this is a control for how quickly to spread water during mopping, or just feedback on how much is left in the tank.

Reply:
the 'volume' DP is actually the speaker volume

Reply:
So I think it can be a number entity then.